pkg/operator/pool/controller.go

```diff
@@ -17,46 +17,110 @@ Some of the code below came from https://github.com/coreos/etcd-operator
 which also has the apache 2.0 license.
 */
 
-// Package cluster to manage a rook cluster.
-package cluster
+// Package pool to manage a rook pool.
+package pool
```

Comment:
Aside from the type, with CRD, you also need a controller for watching, registration with the api extension and resouce creation. So i thought it is cleaner to have all the CRD based files under the same package.